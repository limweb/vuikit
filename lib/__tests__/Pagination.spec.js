import { createVue, destroyVM, toJSON } from './_util'
import waitForUpdate from './_wait-for-update'

describe.skip('Pagination', () => {
  let vm
  afterEach(() => {
    destroyVM(vm)
  })

  it('renders correctly', () => {
    vm = createVue(`
      <vk-pagination :activePage="1" :total="200" compact align="left" />
    `)
    expect(toJSON(vm)).toMatchSnapshot()
    destroyVM(vm)

    vm = createVue(`
      <vk-pagination :activePage="10" :total="200" />
    `)
    expect(toJSON(vm)).toMatchSnapshot()
    destroyVM(vm)

    vm = createVue(`
      <vk-pagination :activePage="20" :total="200" />
    `)
    expect(toJSON(vm)).toMatchSnapshot()
    destroyVM(vm)

    vm = createVue(`
      <vk-pagination :total="200" :limit="10" :pageRange="3" />
    `)
    expect(toJSON(vm)).toMatchSnapshot()
    destroyVM(vm)

    vm = createVue(`
      <vk-pagination :total="100" :limit="1" />
    `)
    expect(toJSON(vm)).toMatchSnapshot()
    destroyVM(vm)
  })

  it('triggers event change when a not active page is clicked', done => {
    vm = createVue(`
      <vk-pagination :total="200" :limit="10" :activePage="1" />
    `, true)

    const cb = jest.fn()
    const pagination = vm.$children[0]

    pagination.$on('change', cb)
    vm.$el.querySelectorAll('li a')[4].click()
    waitForUpdate(() => {
      const args = cb.mock.calls[0]
      const { page } = args[0]
      expect(cb).toHaveBeenCalled()
      expect(args).toHaveLength(1)
      expect(page).toEqual(20)
    }).then(done)
  })
})
