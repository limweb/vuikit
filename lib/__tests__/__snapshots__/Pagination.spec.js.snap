exports[`Pagination renders correctly 1`] = `
Object {
  "child": Array [
    Object {
      "attr": Object {
        "class": Array [
          "uk-pagination",
          "uk-pagination-left",
        ],
      },
      "child": Array [
        Object {
          "attr": Object {
            "class": "uk-disabled",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "span",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "node": "comment",
          "text": "",
        },
        Object {
          "attr": Object {
            "class": "uk-active",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "1",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "...",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon-angle-right",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "a",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon-angle-double-right",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "a",
            },
          ],
          "node": "element",
          "tag": "li",
        },
      ],
      "node": "element",
      "tag": "ul",
    },
  ],
  "node": "root",
}
`;

exports[`Pagination renders correctly 2`] = `
Object {
  "child": Array [
    Object {
      "attr": Object {
        "class": "uk-pagination",
      },
      "child": Array [
        Object {
          "attr": Object {
            "class": "uk-pagination-previous",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon-angle-double-left",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "a",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon-angle-left",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "a",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "...",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "uk-active",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "10",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "...",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "uk-pagination-next",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon-angle-right",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "a",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon-angle-double-right",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "a",
            },
          ],
          "node": "element",
          "tag": "li",
        },
      ],
      "node": "element",
      "tag": "ul",
    },
  ],
  "node": "root",
}
`;

exports[`Pagination renders correctly 3`] = `
Object {
  "child": Array [
    Object {
      "attr": Object {
        "class": "uk-pagination",
      },
      "child": Array [
        Object {
          "attr": Object {
            "class": "uk-pagination-previous",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon-angle-double-left",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "a",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon-angle-left",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "a",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "...",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "uk-active",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "20",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "node": "comment",
          "text": "",
        },
        Object {
          "attr": Object {
            "class": Array [
              "uk-disabled",
              "uk-pagination-next",
            ],
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "span",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
      ],
      "node": "element",
      "tag": "ul",
    },
  ],
  "node": "root",
}
`;

exports[`Pagination renders correctly 4`] = `
Object {
  "child": Array [
    Object {
      "attr": Object {
        "class": "uk-pagination",
      },
      "child": Array [
        Object {
          "attr": Object {
            "class": Array [
              "uk-disabled",
              "uk-pagination-previous",
            ],
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "span",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "node": "comment",
          "text": "",
        },
        Object {
          "attr": Object {
            "class": "uk-active",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "1",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "...",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "uk-pagination-next",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon-angle-right",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "a",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon-angle-double-right",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "a",
            },
          ],
          "node": "element",
          "tag": "li",
        },
      ],
      "node": "element",
      "tag": "ul",
    },
  ],
  "node": "root",
}
`;

exports[`Pagination renders correctly 5`] = `
Object {
  "child": Array [
    Object {
      "attr": Object {
        "class": "uk-pagination",
      },
      "child": Array [
        Object {
          "attr": Object {
            "class": Array [
              "uk-disabled",
              "uk-pagination-previous",
            ],
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "span",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "node": "comment",
          "text": "",
        },
        Object {
          "attr": Object {
            "class": "uk-active",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "1",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "node": "text",
                  "text": "...",
                },
              ],
              "node": "element",
              "tag": "span",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "",
          },
          "child": Array [
            Object {
              "node": "comment",
              "text": "",
            },
          ],
          "node": "element",
          "tag": "li",
        },
        Object {
          "attr": Object {
            "class": "uk-pagination-next",
          },
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon-angle-right",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "a",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-icon-angle-double-right",
                  },
                  "node": "element",
                  "tag": "i",
                },
              ],
              "node": "element",
              "tag": "a",
            },
          ],
          "node": "element",
          "tag": "li",
        },
      ],
      "node": "element",
      "tag": "ul",
    },
  ],
  "node": "root",
}
`;
